api/envoy/config/core/v3/health_check.proto

```diff
@@ -54,6 +54,11 @@ enum HealthStatus {
   DEGRADED = 5;
 }
 
+message HealthStatusSet {
```

Comment:
Do you expect this message to have more fields?
If not, what's the advantage over just specifying the repeated field directly (in your specific case renaming the field to `override_host_status_set`)?

Reply:
This is because we need to distinguish between the case where no override_host_status is configured at all and the case where 0 items are configured. The former will provide a default value.